lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebar.test.tsx

```diff
@@ -5,98 +5,200 @@
-      toggleCloseIcon: faBiohazard,
-      onToggleClick: toggleOpenSpy,
-      logoImg: 'logoImg',
-      logoAltText: 'alt text',
-      logoLink: '/ref'
-    };
-    getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxGlobalSidebar, minimalProps);
-  });
+  const minimalProps: Props = {
+    isOpen: false,
+    toggleOpenIcon: faCrow,
+    toggleCloseIcon: faBiohazard,
+    onToggleClick: () => {},
+    logoImg: 'logoImg',
+    logoAltText: 'alt text',
+    logoLink: '/ref'
+  };
+
+  const quickRender = rtlRender<Props>(NxGlobalSidebar, minimalProps);
 
-  it('renders an aside with nx-global-sidebar class', function() {
-    expect(getShallowComponent()).toMatchSelector('aside.nx-global-sidebar');
+  it('renders a top-level element with role="group"', function() {
+    const view = quickRender(),
+        group = view.getByRole('group');
+
+    expect(group).toHaveAccessibleName('global sidebar');
+    expect(group).toBe(view.container.firstElementChild);
   });
 
-  it('adds classes based on isOpen state', function() {
-    expect(getShallowComponent({ isOpen: true })).toMatchSelector('.nx-global-sidebar.open');
-    expect(getShallowComponent({ isOpen: false })).toMatchSelector('.nx-global-sidebar.closed');
+  describe('children', function() {
+    describe('when only the NxGlobalSidebarNavigation is a provided as a child', function () {
+      it('renders an element with role="navigation" within an element with role="complementary"', function() {
+        const children =
+          <NxGlobalSidebarNavigation key="1">
+            <NxGlobalSidebarNavigationLink icon={faCrow} text="testLink" href="testLink"/>
+          </NxGlobalSidebarNavigation>,
+            view = quickRender({ ...minimalProps, children }),
+            group = view.getByRole('group'),
+            complementary = within(group).getByRole('complementary'),
+            nav = within(complementary).getByRole('navigation');
+
+        expect(group).toBeInTheDocument();
+        expect(complementary).toBeInTheDocument();
+        expect(complementary).toBe(group.firstElementChild);
+        expect(nav).toBeInTheDocument();
+      });
+    });
+
+    describe('when only the NxGlobalSidebarFooter is a provided as a child', function () {
+      it('renders a footer with role="contentinfo"', function() {
+        const children =
+          <NxGlobalSidebarFooter key="1"
+                                 supportText="Support for RSC"
+                                 supportLink="https://github.com/sonatype/sonatype-react-shared-components"
+                                 releaseText="Release 3.1.4"
+                                 productTagLine="Powered by PLAID VILLAIN" />,
+            view = quickRender({ ...minimalProps, children }),
+            group = view.getByRole('group'),
+            footer = within(group).getByRole('contentinfo');
+
+        expect(group).toBeInTheDocument();
+        expect(footer).toBeInTheDocument();
+      });
```

Comment:
Maybe confirm that the footer is not inside the `complementary`

Reply:
> Maybe confirm that the footer is not inside the `complementary`

Good point. Added here: 32fd0c4c07b15486ce3d783467113284475e9dc6